packages/kit/src/runtime/server/index.js

```diff
@@ -260,7 +261,11 @@ export async function respond(request, options, state) {
 												/** @type {import('types').JSONObject} */
 												const data = {};
 												for (let j = 0; j < i; j += 1) {
-													Object.assign(data, await promises[j]);
+													const parent = await promises[j];
+													if (!parent || parent instanceof HttpError || 'error' in parent) {
+														return;
```

Comment:
Is this correct? You just `return undefined`? You don't want to `break` or `return data`?

Reply:
I just changed this to `return data`. I don't think it makes a difference in the end because everything below an error is discarded, but it makes TS shut up, so ..

Reply:
actually you know what? we should be throwing in that case

Reply:
hang on, we're making this way too complicated. the try-catch is in the wrong place. one sec...

Reply:
cancel that, i confused myself — forgot we need to return `[{ data }, { data }, { error }]` to render the correct `+error.svelte` page